pkg/await/fixtures/pod.go

```diff
@@ -61,3 +61,140 @@ func PodBasic() *podBasic {
 		},
 	}
 }
+
+func PodBase(name, namespace string) *corev1.Pod {
+	return &corev1.Pod{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "nginx",
+					Image: "nginx:1.15-alpine",
+				},
+			},
+		},
+		Status: corev1.PodStatus{
+			QOSClass: corev1.PodQOSBurstable,
+		},
+	}
+}
+
+func PodInitialized(name, namespace string) *corev1.Pod {
+	pod := PodBase(name, namespace)
+	pod.Status = corev1.PodStatus{
+		Phase: corev1.PodPending,
+		Conditions: []corev1.PodCondition{
+			{
+				Type:   corev1.PodInitialized,
+				Status: corev1.ConditionTrue,
+			},
+		},
+	}
+
+	return pod
+}
+
+func PodUninitialized(name, namespace string) *corev1.Pod {
+	pod := PodBase(name, namespace)
+	pod.Status = corev1.PodStatus{
+		Phase: corev1.PodPending,
+		Conditions: []corev1.PodCondition{
+			{
+				Type:   corev1.PodInitialized,
+				Status: corev1.ConditionFalse,
+			},
+		},
+	}
+
+	return pod
+}
+
+func PodReady(name, namespace string) *corev1.Pod {
+	pod := PodBase(name, namespace)
+	pod.Status = corev1.PodStatus{
+		Phase: corev1.PodRunning,
+		Conditions: []corev1.PodCondition{
+			{
+				Type:   corev1.PodInitialized,
+				Status: corev1.ConditionTrue,
+			},
+			{
+				Type:   corev1.PodReady,
+				Status: corev1.ConditionTrue,
+			},
+			{
+				Type:   corev1.PodScheduled,
+				Status: corev1.ConditionTrue,
+			},
+		},
+	}
+
+	return pod
+}
+
+func PodSucceeded(name, namespace string) *corev1.Pod {
+	pod := PodBase(name, namespace)
+	pod.Spec.RestartPolicy = corev1.RestartPolicyNever
+	pod.Status = corev1.PodStatus{
+		Phase: corev1.PodSucceeded,
+		Conditions: []corev1.PodCondition{
+			{
+				Type:   corev1.PodInitialized,
+				Status: corev1.ConditionTrue,
+			},
+			{
+				Type:   corev1.PodReady,
+				Status: corev1.ConditionFalse,
```

Comment:
I think this is meant to be `ConditionTrue` for `PodSucceeded`

Reply:
This is actually correct. `PodSucceeded` is a Pod that runs a command and then exits, hence the false Ready condition. I'll add docs to clear that up.